### PR TITLE
[JUJU-779] Update Azure SDK to latest version 62.2

### PR DESCRIPTION
**Pull request by wallyworld**
Update the Azure SDK to version 62.2
This brings in newer APIs for things like pricing info.

NB for compute API, we need to use version `2021-11-01` since `2021-12-01` doesn't appear to be supported yet in all regions.

## QA steps

```sh
juju bootstrap azure \
  --no-default-model \
  --storage-pool name=foo \
  --storage-pool type=azure \
  --storage-pool encrypted=true \
  --storage-pool vault-name-prefix=secret \
  --bootstrap-constraints="root-disk-source=foo"
```

### CODE REVIEW

File: provider/azure/environ.go [hunk 1]
```diff
@@ -1761,8 +1767,16 @@ func (env *azureEnviron) allQueuedInstances(
...
 		deployment := deploymentsResult.Value()
 		deployProvisioningState := resources.ProvisioningStateNotSpecified
+		deployError := "Failed"
 		if deployment.Properties != nil {
 			deployProvisioningState = deployment.Properties.ProvisioningState
+			deployError = string(deployProvisioningState)
+			if deployment.Properties.Error != nil {
+				deployError = to.String(deployment.Properties.Error.Message)
+				if deployment.Properties.Error.Details != nil && len(*deployment.Properties.Error.Details) > 0 {
+					deployError = to.String((*deployment.Properties.Error.Details)[0].Message)
```
Comment (hpidcock) [inline provider/azure/environ.go:1777]: why can there be more than 1 error details and we only care about the first one? this feels like we are not seeing the full error

**Review by hpidcock — APPROVED** [review]

File: provider/azure/environ.go (hunk 1 above)
Comment (wallyworld) [reply provider/azure/environ.go:1777]: cause the first one is the generic "Deployment failed" and the second is the real reason, and in the cases I've seen, there's nothing else beyond that

Comment (wallyworld) [reply provider/azure/environ.go:1777]: We could append any subsequent messages. That tends to lead to these really long chunks of text in status. I can tweak it if needed in the followup work dealing with instance type selection.

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**